### PR TITLE
docs: add operator guide docs

### PR DESCRIPTION
**Pull request by jm-clius**
This PR adds an initial set of guide docs for nwaku operators. The idea is for this body of linked .md-files to grow into a full and self-sufficient guide for nwaku operators and users.

For now the docs are in the main source code repo, but in future this guide must be hosted on an easily-reachable domain. The different domains for Waku documentation is still under discussion.

## What must be reviewed?

Please review the scope, structure and content of the current guides. Also feel free to suggest additional guides/how-to pages, although these will be added in future PRs.

## Where do I start?

The "landing page" is meant to be [`operators/intro.md`](https://github.com/status-im/nwaku/blob/91257aec9978fe110e116297f07ecb723e2f2292/docs/operators/intro.md)

### CODE REVIEW

**Review by oskarth — APPROVED** [review]
Oh wow, this is very thorough, nice work! I looked through it and didn't find anything immediately wrong or missing. I suspect the best way to improve this, to the extent it is needed, will be people using instructions and then asking questions / noticing things.

**Comment by oskarth** [comment]
Might be worth checking with Nimbus ppl who wrote nimbus.guide in terms of things that are useful for beginners here (assuming you haven't already)

**Comment by jm-clius** [comment]
> Oh wow, this is very thorough, nice work! I looked through it and didn't find anything immediately wrong or missing. I suspect the best way to improve this, to the extent it is needed, will be people using instructions and then asking questions / noticing things.

Agree. Had some conversations with new operators who were very helpful. One thing that came up more than once is having some guidance on hardware requirements and getting nwaku to run on different operator architectures (need to mention @PascalPrecht here re discussions about Waku and Raspberry Pi at offsite :)), which we still lack.

> Might be worth checking with Nimbus ppl who wrote nimbus.guide in terms of things that are useful for beginners here (assuming you haven't already)

Have relied extensively on nimbus.guide for inspiration, structure and typical content. :) Good idea to also ping some more people!

File: docs/operators/how-to/run.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+The Discovery v5 routing table can similarly be bootstrapped using a static ENR.
+We include an example below.
+
+```sh
+./build/wakunode2 \
+  --ports-shift:1 \
+  --staticnode:/ip4/0.0.0.0/tcp/60002/p2p/16Uiu2HAkzjwwgEAXfeGNMKFPSpc6vGBRqCdTLG5q3Gmk2v4pQw7H \
+  --staticnode:/ip4/0.0.0.0/tcp/60003/p2p/16Uiu2HAmFBA7LGtwY5WVVikdmXVo3cKLqkmvVtuDu63fe8safeQJ \
+  --discv5-discovery:true \
+  --discv5-bootstrap-node:enr:-JK4QM2ylZVUhVPqXrqhWWi38V46bF2XZXPSHh_D7f2PmUHbIw-4DidCBnBnm-IbxtjXOFbdMMgpHUv4dYVH6TgnkucBgmlkgnY0gmowhCJ6_HaJc2VjcDI1NmsxoQM06FsT6EJ57mzR_wiLu2Bz1dER2nUFSCpaFzCccQtnhYN0Y3CCdl-DdWRwgiMohXdha3UyDw
```
[the remark this answers is not in the document]
Comment (kaiserd) [reply docs/operators/how-to/run.md:86]: Could also link to the how-to/connect page.

**Review by kaiserd — APPROVED** [review]
Very nice! :+1: :100: This is really helpful. Also for new developers!

File: docs/operators/how-to/build.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+This will create a `wakunode2` binary in the `./build/` directory.
+
+> **Note:** Building `wakunode2` requires 2GB of RAM.
+The build will fail on systems not fulfilling this requirement. 
+
+> Setting up a `wakunode2` on the smallest [digital ocean](https://docs.digitalocean.com/products/droplets/how-to/) droplet, you can either
+> * compile on a stronger droplet featuring the same CPU architecture and downgrade after compiling, or
+> * activate swap on the smallest droplet, or
+> * use Docker.
```
Comment (D4nte) [inline docs/operators/how-to/build.md:59]: Maybe a link to the docker section of `run.md`?

File: docs/operators/how-to/run.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
+# Running nwaku
+
+Nwaku binaries can be built and run on Linux and macOS.
```
Comment (D4nte) [inline docs/operators/how-to/run.md:3]: Link to the build doc might be useful

File: docs/operators/how-to/run.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+- listen for incoming libp2p connections on the default TCP port (`60000`)
+- enable `relay` protocol
+- subscribe to the default pubsub topic, namely `/waku/2/default-waku/proto`
+- enable `store` protocol, but only as a client.
+This implies that the nwaku node will not persist any historical messages itself,
+but can query `store` service peers who do so.
+To configure `store` as a service node,
+see [this tutorial](./configure-store.md).
+
+> **Note:** The commonly used `filter` and `lightpush` protocols are _not_ enabled by default.
```
Comment (D4nte) [inline docs/operators/how-to/run.md:28]:
```suggestion
> **Note:** The `filter` and `lightpush` protocols are _not_ enabled by default.
```
How common are they ? :)

File: docs/operators/how-to/run.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+Consult the [configuration guide](./configure.md) on how to configure your nwaku node to run these protocols.
+
+Some typical non-default configurations are explained below.
+For more advanced configuration, see the [configuration guide](./configure.md).
+
+## Finding your listening address(es)
+
+Find the log entry beginning with `Listening on`.
```
Comment (D4nte) [inline docs/operators/how-to/run.md:36]: Might be interesting to provide an alternative using the JSON RPC with `curl`.

**Review by D4nte — APPROVED** [review]

File: docs/operators/how-to/build.md (hunk 2 above)
Comment (LNSD) [inline docs/operators/how-to/build.md:59]: For me, this "use Docker" is a little ambiguous 🤔 Does it mean "Use docker to build it"? In that case, I am missing the "building with docker" instructions link if we have that possibility.

File: docs/operators/how-to/run.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/how-to/run.md:152]: Done and done in [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/how-to/configure-domain.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+It is possible to configure an IPv4 DNS domain name that resolves to the node's public IPv4 address.
+
+```shell
+wakunode2 --dns4-domain-name=mynode.example.com
+```
+
+This allows for the node's publically announced `multiaddrs` to use the `/dns4` scheme.
+In addition, nodes with domain name and [secure websocket configured](./configure-websocket.md),
+will generate a discoverable ENR containing the `/wss` multiaddr with `/dns4` domain name.
+This is necessary to verify domain certificates when connecting to this node over secure websocket.
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/how-to/configure-domain.md:16]: Good point! Have added notes on both pages to clarify that these are independent. [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/how-to/configure-store.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+```sh
+./build/wakunode2 \
+  --store:true \
+  --storenode:/dns4/node-01.ac-cn-hongkong-c.wakuv2.test.statusim.net/tcp/30303/p2p/16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm
+```
+
+You can now send queries to retrieve historical messages
+from the configured store service node
+using the [Waku v2 JSON RPC API](https://rfc.vac.dev/spec/16/).
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/how-to/configure-store.md]: Thanks! Much better. [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/how-to/configure-store.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+## Configuring a store service node
+
+To store historical messages on your node which can be served to store clients the `--persist-messages` CLI option must be enabled.
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/how-to/configure-store.md:34]: Fully agree on both counts. I think we need to rethink store config in general now that we're changing its implementation and will require time-based configuration and more. I think we should also make it possible to run a `store` client without ever mounting store protocol.

File: docs/operators/how-to/configure-websocket.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
+# Configure websocket transport
+
+Websocket is currently the only Waku transport supported by browser nodes that uses [js-waku](https://github.com/status-im/js-waku).
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/how-to/configure-websocket.md]: Changed to `using`.

File: docs/operators/intro.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
+# The nwaku guide for operators
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/intro.md]: Done in [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/intro.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+connect to an existing Waku v2 network
+and use existing tools for monitoring and maintaining a running node.
+
+## Helpful resources
+
+<!-- TODO -->
+
+## Getting in touch or reporting an issue
+
+For an inquiry, or if you would like to propose new features, feel free to [open a general issue](https://github.com/status-im/nim-waku/issues/new/).
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/intro.md]: [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/intro.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+For bug reports, please [tag your issue with the `bug` label](https://github.com/status-im/nim-waku/issues/new/).
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply docs/operators/intro.md]: [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/how-to/build.md (hunk 2 above)
Comment (jm-clius) [reply docs/operators/how-to/build.md:59]: [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/how-to/run.md (hunk 3 above)
Comment (jm-clius) [reply docs/operators/how-to/run.md]: [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/how-to/run.md (hunk 4 above)
Comment (jm-clius) [reply docs/operators/how-to/run.md]: Hahahaha. Not common enough, if you ask me. Fair point. Have changed wording in [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

File: docs/operators/how-to/run.md (hunk 5 above)
Comment (jm-clius) [reply docs/operators/how-to/run.md:37]: Done in [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2)

**Comment by jm-clius** [comment]
Thanks for your comments, everyone. Have addressed all initial comments in [6cb52dc](https://github.com/status-im/nwaku/pull/963/commits/6cb52dcfb28d0b68a1564e7e127172633e056ab2). We can expand this body of docs as we continue.

**Review by LNSD — APPROVED** [review]
This is a great starting point. It is also helpful for my onboarding. 

Great job 😁

File: docs/operators/how-to/build.md (hunk 2 above)
Comment (jm-clius) [reply docs/operators/how-to/build.md:59]: No, you're quite right. We don't yet have Docker as part of our instruction manual. It is used by our different fleet deployments, though, and images are published by those same deployment script to Dockerhub. Many people have figured out where to get those images or how to use our Dockerfile to compile their own containers. We definitely need an authoratitve guide, though.

**Pull request merged**